### PR TITLE
[PIL-757] - Add RFM Secondary Contact Details

### PR DESCRIPTION
**Pull request by aashik-varahan**

### CODE REVIEW

File: app/controllers/rfm/RfmAddSecondaryContactController.scala [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import controllers.actions._
+import forms.RfmAddSecondaryContactFormProvider
+import models.Mode
+import navigation.ReplaceFilingMemberNavigator
+import pages.{RfmAddSecondaryContactPage, RfmPrimaryContactEmailPage, RfmPrimaryContactNamePage}
+import play.api.i18n.I18nSupport
+import play.api.libs.json.Format.GenericFormat
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import play.api.mvc.Result
```
Comment (jassalrichy) [inline app/controllers/rfm/RfmAddSecondaryContactController.scala:30]: Remove unused import.

File: test/controllers/rfm/RfmAddSecondaryContactControllerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+class RfmAddSecondaryContactControllerSpec extends SpecBase {
+
+  val formProvider = new RfmAddSecondaryContactFormProvider()
+
+  "RfmAddSecondaryContact Controller" when {
+
+    "must return OK and the correct view for a GET" in {
+      val userAnswers = UserAnswers(userAnswersId)
+        .setOrException(RfmPrimaryContactNamePage, "name")
+        .setOrException(RfmPrimaryContactEmailPage, "asda")
```
Comment (jassalrichy) [inline test/controllers/rfm/RfmAddSecondaryContactControllerSpec.scala:43]: We should not be using test data such as an actual company name "asda" (which looks like PII data). Suggest we change this to something like "john.doe@example.com"

File: test/controllers/rfm/RfmAddSecondaryContactControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(formProvider(), "name", NormalMode)(request, appConfig(application), messages(application)).toString
+      }
+    }
+
+    "must populate the view correctly on a GET when the question has previously been answered" in {
+
+      val userAnswers = UserAnswers(userAnswersId)
+        .setOrException(RfmPrimaryContactNamePage, "name")
+        .setOrException(RfmPrimaryContactEmailPage, "asda")
```
Comment (jassalrichy) [inline test/controllers/rfm/RfmAddSecondaryContactControllerSpec.scala:63]: See comment above.

File: app/controllers/rfm/RfmAddSecondaryContactController.scala (hunk 1 above)
Comment (aashik-varahan) [reply app/controllers/rfm/RfmAddSecondaryContactController.scala]: Done.

File: test/controllers/rfm/RfmAddSecondaryContactControllerSpec.scala (hunk 2 above)
Comment (aashik-varahan) [reply test/controllers/rfm/RfmAddSecondaryContactControllerSpec.scala]: Changed as suggested.

File: test/controllers/rfm/RfmAddSecondaryContactControllerSpec.scala (hunk 3 above)
Comment (aashik-varahan) [reply test/controllers/rfm/RfmAddSecondaryContactControllerSpec.scala]: Changed as suggested.

**Review by jassalrichy — APPROVED** [review]

**Review by javed-iqbal1 — APPROVED** [review]

**Pull request merged**